lib/erb_lint/linters/hard_coded_string.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'better_html/tree/tag'
+
+module ERBLint
+  module Linters
+    # Checks for hardcoded strings. Useful if you want to ensure a string can be translated using i18n.
+    class HardCodedString < Linter
+      include LinterRegistry
+
+      class ConfigSchema < LinterConfig
+        property :present, accepts: [true, false], default: true, reader: :present?
+      end
+      self.config_schema = ConfigSchema
+
+      def offenses(processed_source)
+        hardcoded_strings = processed_source.ast.descendants(:text).each_with_object([]) do |text_node, to_check|
+          next if javascript?(processed_source, text_node)
+
+          text_node.to_a.each do |inner_node|
+            if relevant_node?(inner_node)
+              to_check << text_node
+            end
+          end
+        end
+
+        hardcoded_strings.compact.map do |string|
+          Offense.new(
+            self,
+            processed_source.to_source_range(string.loc.start, string.loc.stop),
+            "String not translated: #{string.loc.source.lstrip}"
+          )
+        end
+      end
+
+      private
+
+      def javascript?(processed_source, text_node)
+        ast = processed_source.parser.ast.to_a
+        index = ast.find_index(text_node)
+
+        previous_node = ast[index - 1]
+
+        if previous_node.type == :tag
+          tag = BetterHtml::Tree::Tag.from_node(previous_node)
+
+          tag.name == "script"
+        end
+      end
+
+      def relevant_node?(inner_node)
+        source = ''
+
+        if inner_node.is_a?(String)
+          source = inner_node
+        elsif inner_node.type == :text
```

Comment:
As the parser is written now there will not be nested `:text` nodes, so this branch won't do anything. A `s(:text)` node can only contain children that are `String` or `s(:erb)` nodes.